capstone/src/test/java/com/google/sps/servlets/CheckNewUserServletTest.java

```diff
@@ -0,0 +1,136 @@
+  private static final String OTHER_USER_ID = "testytwo-mc-test";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+            new LocalDatastoreServiceTestConfig()
+                .setDefaultHighRepJobPolicyUnappliedJobPercentage(0),
+            new LocalUserServiceTestConfig())
+          .setEnvEmail(CURRENT_USER_EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail.com")
+          .setEnvAttributes(
+              new HashMap(
+                  ImmutableMap.of(
+                      "com.google.appengine.api.users.UserService.user_id_key", CURRENT_USER_ID)));
+
+  private static final User CURRENT_USER = new User(CURRENT_USER_ID, "Test", "McTest", 
+                          CURRENT_USER_EMAIL, 
+                          /* phoneNumber= */ "123-456-7890", 
+                          /* profilePic= */ "", 
+                          /* badges= */ new LinkedHashSet<Badge>(), 
+                          /* groups= */ new LinkedHashSet<Long>(), 
+                          /* interests= */ new ArrayList<String>());
+
+  private static final User OTHER_USER = new User(OTHER_USER_ID, "Test Two", "McTest", 
+                          OTHER_USER_EMAIL, 
+                          /* phoneNumber= */ "123-456-7890", 
+                          /* profilePic= */ "", 
+                          /* badges= */ new LinkedHashSet<Badge>(), 
+                          /* groups= */ new LinkedHashSet<Long>(), 
+                          /* interests= */ new ArrayList<String>());                        
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private StringWriter responseWriter;
+  private DatastoreService datastore;
+  private CheckNewUserServlet checkNewUserServlet;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+
+    addUserToDatastore(datastore, OTHER_USER);
+
+    // Set up a fake HTTP response.
+    responseWriter = new StringWriter();
+    when(mockResponse.getWriter()).thenReturn(new PrintWriter(responseWriter));
+
+    checkNewUserServlet = new CheckNewUserServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doPost_newUserVisiting() throws Exception {
+    checkNewUserServlet.doGet(mockRequest, mockResponse);
+    String response = responseWriter.toString();
+    assertThat(response).contains(/* isUserNew= */ "true");
+  }
+
+  @Test
+  public void doPost_existingUserVisiting() throws Exception {
+    addUserToDatastore(datastore, CURRENT_USER);
+
+    checkNewUserServlet.doGet(mockRequest, mockResponse);
+    String response = responseWriter.toString();
+    assertThat(response).contains(/* isUserNew= */ "false");
+
+    removeUserFromDatastore(datastore, CURRENT_USER);
```

Comment:
You shouldn't need to remove the user that was added to the DB at the end of the test because the helper.tearDown() should do that for you. Also, any post processing should be done in the tearDown function instead of after the asserts